__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -0,0 +1,1665 @@
+
+    // Validating getAddressAtIndex and getAllAddresses methods
+    for (let i = 0; i < 22; ++i) {
+      // Validating generator results
+      const genResults = await addressGenerator.next();
+      expect(genResults).toHaveProperty('value');
+      expect(genResults).toHaveProperty('done');
+
+      // Validating gap limit
+      if (i === 21) {
+        expect(genResults.done).toEqual(true);
+        expect(genResults.value).toBeUndefined();
+        break;
+      }
+
+      // Validating generator contents
+      const addressAtIndex = hWallet.getAddressAtIndex(i);
+      expect(genResults.value.index).toEqual(i);
+      expect(genResults.value.address).toEqual(addressAtIndex);
+    }
+
+    // Validating currentAddress behavior
+    let currentAddress = hWallet.getCurrentAddress();
+    expect(currentAddress.index).toEqual(0);
+    expect(currentAddress.address).toEqual(hWallet.getAddressAtIndex(0));
+    // Expect no change on second call
+    currentAddress = hWallet.getCurrentAddress();
+    expect(currentAddress.index).toEqual(0);
+    expect(currentAddress.address).toEqual(hWallet.getAddressAtIndex(0));
+    // Expect the same address for the last time when calling with markAsUsed parameters
+    currentAddress = hWallet.getCurrentAddress({ markAsUsed: true });
+    expect(currentAddress.index).toEqual(0);
+    expect(currentAddress.address).toEqual(hWallet.getAddressAtIndex(0));
+    // Now it won't return the used one
+    currentAddress = hWallet.getCurrentAddress();
+    expect(currentAddress.index).toEqual(1);
+    expect(currentAddress.address).toEqual(hWallet.getAddressAtIndex(1));
+
+    // Validating getNextAddress behavior
+    let nextAddress = hWallet.getNextAddress();
+    expect(nextAddress.index).toEqual(2);
+    expect(nextAddress.address).toEqual(hWallet.getAddressAtIndex(2));
+    // Expecting the next address index
+    nextAddress = hWallet.getNextAddress();
+    expect(nextAddress.index).toEqual(3);
+    expect(nextAddress.address).toEqual(hWallet.getAddressAtIndex(3));
+
+    // Expect the "current address" to change when a transaction arrives at the current one
+    currentAddress = hWallet.getCurrentAddress();
+    await GenesisWalletHelper.injectFunds(currentAddress.address, 1);
+    const currentAfterTx = hWallet.getCurrentAddress();
+    expect(currentAfterTx.index).toEqual(currentAddress.index + 1);
+    expect(currentAfterTx.address).toEqual(hWallet.getAddressAtIndex(currentAddress.index + 1));
+  });
+});
+
+describe('getTransactionsCountByAddress', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should return correct entries for a wallet', async () => {
+    // Create the wallet
+    const hWallet = await generateWalletHelper();
+
+    // Validate empty contents, properties with the address string as a key
+    const tcbaEmpty = hWallet.getTransactionsCountByAddress();
```

Comment:
`tcba` is meant to be **T**ransactions **C**ount **B**y **A**ddress, in a more concise way of writing.
Empty because this wallet is pristine and doesn't have any transactions.